### PR TITLE
Feat/exploit rework

### PR DESCRIPTION
**Pull request by mariotee**
also includes https://github.com/SWU-Karabast/SWUOnline/pull/423

### CODE REVIEW

File: DecisionQueue/DecisionQueueEffects.php [hunk 1]
```diff
@@ -618,11 +618,11 @@ function SpecificCardLogic($player, $card, $lastResult)
       }
       break;
     case "COUNTDOOKU_TWI":
-      $power = CardPower($lastResult);
-      AddDecisionQueue("MULTIZONEINDICES", $player, "MYALLY&THEIRALLY", 1);
-      AddDecisionQueue("SETDQCONTEXT", $player, "Choose a unit to deal " . $power . " damage to", 1);
-      AddDecisionQueue("CHOOSEMULTIZONE", $player, "<-", 1);
-      AddDecisionQueue("MZOP", $player, "DEALDAMAGE," . $power, 1);
+        $power = $lastResult;
```
Comment (VJubert) [inline DecisionQueue/DecisionQueueEffects.php:621]: indent seems wrong

File: AllyAbilities.php [hunk 2]
```diff
@@ -1232,7 +1248,7 @@ function AllyPlayCardAbility($cardID, $player="", $from="-", $abilityID="-", $un
         AddDecisionQueue("PASSPARAMETER", $player, "MYALLY-" . $index, 1);
         AddDecisionQueue("ADDMZUSES", $player, -1, 1);
       }
-      break;
+      break;
```
Comment (VJubert) [inline AllyAbilities.php:1251]: revert

File: AllyAbilities.php [hunk 3]
```diff
@@ -221,6 +221,13 @@ function DestroyAlly($player, $index, $skipDestroy = false, $fromCombat = false,
     CollectBounties($player, $index);
     IncrementClassState($player, $CS_NumAlliesDestroyed);
   }
+
+  if ($fromExploit) {
```
Comment (VJubert) [inline AllyAbilities.php:225]: why did you have special case for exploit ?

File: DecisionQueue/DecisionQueueEffects.php (hunk 1 above)
Comment (mariotee) [reply DecisionQueue/DecisionQueueEffects.php]: updated

File: AllyAbilities.php (hunk 2 above)
Comment (mariotee) [reply AllyAbilities.php]: reverted

File: AllyAbilities.php (hunk 3 above)
Comment (mariotee) [reply AllyAbilities.php]: i had a different solution before coming to this one. i must have missed this spot in refactoring. that has now been removed

**Review by OotTheMonk — APPROVED** [review]

**Pull request merged**